### PR TITLE
add post a new comment

### PR DESCRIPTION
**Pull request by BBSharps**

### CODE REVIEW

File: src/elements/review_by_id.jsx [hunk 1]
```diff
@@ -61,8 +70,56 @@ function ReviewById() {
...
+          defaultValue="guest"
+          onChange={(event) => {
+            setLogUser(event.target.value);
+          }}
+          name="Users"
+          id="users"
+        >
+          <option>guest</option>
+          {users.map((user) => {
+            return <option>{user.username}</option>;
```
Comment (Poonam-raj) [inline src/elements/review_by_id.jsx]: Please remember to add keys to your maps!

File: src/elements/review_by_id.jsx [hunk 2]
```diff
@@ -61,8 +70,56 @@ function ReviewById() {
...
+                setNewCommentInput(
+                  "Sorry comment failed, please refresh the page"
+                );
+              });
+            setNewCommentInput("");
+          }}
+          id="newComment"
+        >
+          <h2>Submit a new comment as "{logUser}"</h2>
+          <input
```
Comment (Poonam-raj) [inline src/elements/review_by_id.jsx]: Please make this able to take a multiline input (you'll need a different element)

File: src/elements/review_by_id.jsx [hunk 3]
```diff
@@ -61,8 +70,56 @@ function ReviewById() {
...
+          }}
+          name="Users"
+          id="users"
+        >
+          <option>guest</option>
+          {users.map((user) => {
+            return <option>{user.username}</option>;
+          })}
+        </select>
+      </form>
```
Comment (Poonam-raj) [inline src/elements/review_by_id.jsx]: THis could be extracted into its own component

File: src/elements/review_by_id.jsx [hunk 4]
```diff
@@ -61,8 +70,56 @@ function ReviewById() {
...
+          <input
+            value={newCommentInput}
+            onInput={(event) => {
+              setNewCommentInput(event.target.value);
+            }}
+          ></input>
+          <button htmlFor="newComment" type="submit">
+            Submit
+          </button>
+        </form>
```
Comment (Poonam-raj) [inline src/elements/review_by_id.jsx]: This could be its own component

File: src/elements/review_by_id.jsx [hunk 5]
```diff
@@ -61,8 +70,56 @@ function ReviewById() {
...
+      ) : null}
       <ul className="commentsList">
+        {optimisticComment.map((comment) => {
+          return <ReviewCommentCard comment={comment} />;
```
Comment (Poonam-raj) [inline src/elements/review_by_id.jsx]: Please remember keys for maps

File: src/elements/review_by_id.jsx [hunk 6]
```diff
@@ -61,8 +70,56 @@ function ReviewById() {
...
+        })}
         {reviewComments.map((comment) => {
           return <ReviewCommentCard comment={comment} />;
```
Comment (Poonam-raj) [inline src/elements/review_by_id.jsx]: please remember keys for maps

**Review by Poonam-raj — CHANGES_REQUESTED** [review]
Please make sure the user can't post a empty comment

I also want you to make sure a user can't post or click submit when the form is still submitting (you can test this by slowing your internet down in the browser and spamming the submit button when you submit a valid comment)

Make the PR again once you've made changes and ofc let me know if you have qs

File: src/elements/Submit-form.jsx [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+        event.preventDefault();
+        setSubmitting(true);
+        setNewCommentInput("");
+        postComment(number, logUser, newCommentInput)
+          .then((response) => {
+            setOptimisticComment([response, ...optimisticComment]);
+            setSubmitting(false);
+          })
+          .catch((error) => {
+            setNewCommentInput("Sorry comment failed, please refresh the page");
```
Comment (B7Bprog) [inline src/elements/Submit-form.jsx]: This error message didn't make it to the screen.

**Review by B7Bprog — CHANGES_REQUESTED** [review]
Hey Billy, it's looking pretty good, but there's still an issue remaining. When axios throws an error, your error message is not displayed. Instead the submit button reappears without the textarea. (See screenshot below) I recommend to set up a state for errors and use that to render the page conditionally. 
(Regarding your textarea, please disable resizing.)

Once you've fixed these please feel free to merge. 👍 

Screenshot 2023-01-13 at 10 51 03

**Pull request merged**